src/modules/shorthand/add.rs

```diff
@@ -31,22 +31,26 @@ impl SyntaxModule<ParserMetadata> for ShorthandAdd {
     }
 
     fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
-        let var_tok = meta.get_current_token();
         self.var = variable(meta, variable_name_extensions())?;
         token(meta, "+=")?;
-        let variable = handle_variable_reference(meta, &var_tok, &self.var)?;
-        prevent_constant_mutation(meta, &var_tok, &self.var, variable.is_const)?;
-        self.kind = variable.kind;
-        self.global_id = variable.global_id;
-        self.is_ref = variable.is_ref;
         syntax(meta, &mut *self.expr)?;
+        // Variable reference handling moved to typecheck phase
         Ok(())
     }
 }
 
 impl TypeCheckModule for ShorthandAdd {
     fn typecheck(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
         self.expr.typecheck(meta)?;
+        
+        // Handle variable reference in typecheck phase
+        let var_tok = meta.get_current_token(); // Best effort to get current token
```

Comment:
For such case store the positon of this token in the structure like in `rhs_pos` field. Do it in all places where you get current token in `typecheck` phase